x-pack/plugins/lists/server/routes/import_list_item_route.ts

```diff
@@ -25,8 +25,13 @@ export const importListItemRoute = (router: IRouter, config: ConfigType): void =
           accepts: ['multipart/form-data'],
           maxBytes: config.maxImportPayloadBytes,
           parse: false,
+          timeout: config.importTimeout, // payload has 5 minutes to upload before it times out
         },
         tags: ['access:lists-all'],
+        timeout: {
+          server: false, // The server should never not timeout while processing
+          socket: config.importTimeout + 1, // The socket timeout _must_ be greater than the payload timeout, so we add 1 additional millisecond to it
+        },
```

Comment:
Is it really necessary to add all these three options? I mean, I know that technically, these are distinct hapi configuration parameters, however as a user I would probably expect to have only one route configuration option to impact all three at the same time. a kiss `timeout` option.

Realistically, would there be any cases were `parse.timeout` would NOT be ~= `socket.timeout`?

Additional point, is hapi allowing to totally disable server or socket timeout by using `false` or `true`? If that's the case, this is probably something we just don't want to allow in the route configuration API, as it is just too dangerous imho.

Reply:
To me, having all 3 configurations seems to make sense and I wouldn't want to take away the power from people trying to configure their routes. The defaults are what keeps it at KISS level, and I appreciate being able to configure all three independent of each other if I need to so I can decide how to divide up the overall socket time between the handshake and upload phase, then the server side processing and return response phase. 

From my reading and understanding of things ...

The first `body.timeout` option is for when a user posts a payload to make the amount of time to get the full payload before beginning processing the route endpoint configurable. If this is configured by a route then it has to be less than the overall `timeout.socket` timeout.

The second `timeout.server` is for how long you want the connection to last when the server starts processing and is for when you want to limit the server side processing and that also has to be less than the overall `timeout.socket` option. Default of false here is what it is on all the other routes which is why I left it in my route to still be false.

The third, `timeout.socket` is for the overall amount of time from the beginning of the socket being created to the payload upload to the server side processing and response that the socket can live before being forcibly closed. Which I believe would probably include the handshaking, the payload upload processing time, server side processing time, and the response. So you optionally can set this to be a particular overall value such as NodeJS's default of 2 minutes and it will guarantee that the socket can at maximum live that long.

Having all three like this allows me to do the following:

I want the overall maximum amount of time my socket can live starting from handshake phase to server processing to returning from the response to be at most 5 minutes and 1 millisecond, but if the client side wants to it can take up to the full 5 minutes and the back end has whatever time is left over to finish processing before the socket is closed.

So, if the payload takes 4 minutes to upload, the backend has at most 1 minute left over to process the request and return a response before the socket is removed. If the payload takes 2 minutes, then the backend has 3 minutes to do its processing.

If I combined this into just one timeout of say 5 minutes I will lose the ability to divide that 5 minutes up into different slices to decide how much time to allow the payload to upload and how much time to allow the backend to do its processing.

I can see other routes making different decisions if they feel that the amount of time divided up between a payload upload and a server side processing should be a different configurable amount of time or if they want to limit the amount of time for a payload vs. the total amount of time the backend can have to do processing.

---

However, with that being said, I if you want the overall `timeout.socket` to be set to a maximum value of something like 10 minutes that downstream users of routes cannot re-configure without a new pull request to change it to be safe here I can see that. If you limit it, then the other two options of `body.timeout` and `timeout.server` cannot exceed that overall time period. If you want to remove the options of `false` and keep each route's maximum time that of `timeout.socket` of something like 10 minutes that is totally doable.

Reply:
> The defaults are what keeps it at KISS level, and I appreciate being able to configure all three independent of each other if I need to so I can decide how to divide up the overall socket time between the initial handshake/socket create phase,

But right now it's not a requirement for your use-case, right? So we can start with exposing a single `timeout` option and extend route configuration with `timeout.server` & `timeout.socket` when it's necessary.

Reply:
That is true, @restrry, we can can expose a single timeout option and use that for now. The least amount of API surface area to handle my use case would be to expose a single millisecond timeout like so:


```ts
timeout: number
```

I will do that for this PR and that should work out. If we need finer grain controls we can add them later. I will update this PR right now to only expose a single timeout option.